bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -121,12 +122,16 @@ impl Client {
             let mut filter = FilterDefinition::default();
             let mut room_filter = RoomFilter::default();
             let mut event_filter = RoomEventFilter::default();
+            let mut timeline_filter = RoomEventFilter::default();
 
             event_filter.lazy_load_options =
                 LazyLoadOptions::Enabled { include_redundant_members: false };
             room_filter.state = event_filter;
             filter.room = room_filter;
 
+            timeline_filter.limit = limit.and_then(UInt::new);
```

Comment:
Hmm, an u64 doesn't fit into an `UInt`, no? If the `u64` is too large `UInt::new()` returns `None`. This is also where a smaller integer type on the public facing API helps, an `u32` or smaller type fits into `UInt`.

Reply:
I based this on the [docs](https://docs.rs/js_int/latest/js_int/) which say
> This crate provides the types Int and UInt which wrap i64 and u64, respectively.

Seemed easier to not worry about unnecessary conversions if it was backed by a `u64`.

Reply:
Indeed, but they aren't simple wrappers, they have some special limits:

> This crate provides the types Int and UInt which wrap i64 and u64, respectively. These types add bounds checking to ensure the contained value is within the range representable by a JavaScript Number

This means that any conversion from an `u64` to an `UInt` might return an error, or rather an `Option` since the error type is trivial. The docs showcase this here: https://docs.rs/js_int/latest/js_int/struct.UInt.html#examples-2

An `u32` on the other hand can be always converted using a simple [`.into()` call](https://docs.rs/js_int/latest/js_int/struct.UInt.html#impl-From%3Cu32%3E).

If you really would like to use the `u64`, using the [`UInt::new_satturating()`](https://docs.rs/js_int/latest/js_int/struct.UInt.html#method.new_saturating) should marginally make more sense.

Reply:
> An u32 on the other hand can be always converted using a simple [.into() call](https://docs.rs/js_int/latest/js_int/struct.UInt.html#impl-From%3Cu32%3E).

Ah ha! I was just about to go searching for how to do this with a u16 👍

Reply:
`Into::into()` works for all the smaller types, since they are guaranteed to fit into the `UInt`, or rather in the JSON `Number` type.

Reply:
Updated ✅